kod/object/active/holder/nomoveon/battler/monster/avchiefpois.kod

```diff
@@ -0,0 +1,374 @@
+         AND Send(poOwner,@LineOfSight,#obj1=self,#obj2=poTarget)
+         AND Send(poOwner,@ReqSomethingAttack,#what=self,#victim=poTarget,
+                  #use_weapon=self)
+      {
+         Send(self,@DoMissileAttack);
+
+         return TRUE;
+      }
+
+      return FALSE;
+   }
+
+   DoMissileAttack()
+   {
+      pbFire = TRUE;
+      Send(poOwner,@SomethingShot,#who=self,#target=poTarget,#projectile=self);
+      Send(self,@TryAttack,#what=poTarget,#stroke_obj=self);
+
+      return;
+   }
+
+   GetProjectileSpeed()
+   {
+      return 15;
+   }
+
+   SendProjectileAnimation()
+   {
+      AddPacket(1,ANIMATE_NONE,2,1);
+
+      return;
+   }
+
+   GetProjectileIcon()
+   {
+      return AvarChiefPoison_arrow;
+   }
+
+   GetProjectileLightFlags()
+   {
+      % No lighting information.
+      return LIGHT_FLAG_NONE;
+   }
+
+   GetOptimalRange()
+   {
+      return random(8,15);
+   }
+
+   HitSideEffect(what=$)
+   {
+      local oSpell;
+
+      if Random(1,POISON_CHANCE) = 1
+         AND IsClass(what,&Player)
+         AND NOT Send(what,@IsEnchanted,#byClass=&Poison)
+      {
+         oSpell = Send(SYS,@FindSpellByNum,#num=SID_POISON);
+         Send(oSpell,@MakePoisoned,#who=what,
+               #lossrate=POISON_LOSSRATE,#duration=POISON_DURATION);
+
+         if IsClass(what,&Player)
+         {
+            Send(what,@EffectSendUserDuration,#effect=EFFECT_WAVER,
+                  #duration=10000);
+            Send(what,@MsgSendUser,
+                  #message_rsc=AvarChiefPoison_poisoned_add_rsc);
+         }
+      }
+
+      return;
+   }
+
+   IsAlly(target = $)
+   {
+      if IsClass(target,&AvarChieftainPoison)
+      {
+         return TRUE;
+      }
+
+      % Charm checks up higher.
+      propagate;
+   }
+
+   SpecialHatredBehavior(what=$)
+   {
+      % Avars attack other avars of other clans, but only in the
+      %  temple of riija ruins (rid_KC3).
+      if IsClass(what,&AvarShaman)
+         OR IsClass(what,&AvarChieftain)
+         OR IsClass(what,&Avar)
+      {
+         if poOwner <> $ AND Send(poOwner,@GetRoomNum) = RID_KC3
+            AND Send(what,@GetClan) <> Send(self,@GetClan)
+         {
+            return 25;
+         }
+      }
+
+      return 0;
+   }
+
+   GetTreasureType()
+   {
+      return viTreasure_type;
+   }
+
+   CreateTreasure(who=$,corpse=$)
+   {
+      local oTreasure, iNumber;
+
+      % Apparitions, etc, provide no treasure
+      if pbIllusion
+      {
+         return;
+      }
+
+      oTreasure = Create(&NeruditeArrow,#corpse=corpse);
+      iNumber = Random(25,45);
+      Send(oTreasure,@AddNumber,#number=iNumber);
+
+      if Send(poOwner,@ReqNewHold,#what=oTreasure,
+               #new_row=piRow,#new_col=piCol)
+         AND Send(poOwner,@ReqSomethingMoved,#what=oTreasure,
+                  #new_row=piRow,#new_col=piCol)
+      {
+         Send(poOwner,@NewHold,#what=oTreasure,#new_row=piRow,#new_col=piCol);
```

Comment:
NewHold should have fine_row and fine_col parameters (see avchief.kod, this was fixed for all NewHold calls in the past few months and stops loot dropping inside walls).